### PR TITLE
Dhruv/mt5 password modal styling/trah 3963

### PR DESCRIPTION
**Pull request by Dhruv-deriv**
## Changes:

1. To style the new password modal in MT5-new Account creation 
2. To create the Mobile view for the new MT5 Account Creation Password model.
3. Created test cases for Mt5CreatePassword Modal

## Screenshot:

1. Desktop view:
Screenshot 2024-08-29 at 9 55 22 AM

2. Mobile view:
Screenshot 2024-08-29 at 9 55 48 AM

3. Test cases 
Screenshot 2024-08-23 at 2 41 02 PM

### CODE REVIEW

File: packages/appstore/src/components/cfds-listing/cfds-listing.scss [hunk 1]
```diff
@@ -2044,7 +2044,6 @@
             overflow-x: hidden;
             height: 100%;
             max-height: calc(100% - 6.4rem);
-            padding: 1rem;
```
Comment (amina-deriv) [inline packages/appstore/src/components/cfds-listing/cfds-listing.scss]: why is this removed?

File: packages/cfd/src/Containers/cfd-password-modal.tsx [hunk 2]
```diff
@@ -36,11 +36,11 @@ import { useCfdStore } from '../Stores/Modules/CFD/Helpers/useCfdStores';
 import { CFD_PLATFORMS, CATEGORY } from '../Helpers/cfd-config';
 import classNames from 'classnames';
 import { getDxCompanies, getMtCompanies, TDxCompanies, TMtCompanies } from '../Stores/Modules/CFD/Helpers/cfd-config';
-import '../sass/cfd.scss';
+import './mt5-create-password/mt5-create-password.scss';
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/cfd-password-modal.tsx]: why do we need to import mt5-create-password.scss to cfd-password-modal? can't we import it only to mt5-create-password.tsx?

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx [hunk 3]
```diff
@@ -125,10 +130,13 @@ const MT5CreatePassword = ({
                                 onCheck={() => setChecked(prev => !prev)}
                                 need_tnc={need_tnc}
                             />
+                        </div>
+                        <div className='mt5-password-submit-button'>
                             <FormSubmitButton
                                 is_disabled={!values.password || !checked || Object.keys(errors).length > 0}
                                 is_loading={isSubmitting}
                                 label={localize('Create account')}
+                                // class='mt5-password-submit-button'
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx]: lets remove this commented code

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+    }
+    .dc-password-meter {
+        &__bg {
+            background: var(--general-section-2);
+            @include mobile-or-tablet-screen {
+                background: var(--fill-disabled);
+            }
+        }
+    }
+    .status-dialog {
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: where is this used?

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+        margin-top: 3em;
+    }
+    & .dc-password-meter__container {
+        flex-grow: 1;
+        margin: auto;
+
+        @include mobile-or-tablet-screen {
+            // width: calc(100vw - 4.8rem);
+            // max-width: 30rem;
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: remove commented codes

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+        }
+        &:last-child {
+            margin: 0;
+        }
+
+        @include mobile-or-tablet-screen {
+            width: 100%;
+        }
+    }
+    &--absolute {
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: where is this used?

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx [hunk 7]
```diff
@@ -68,15 +68,20 @@ const MT5CreatePassword = ({
...
-                    <div className='mt5_password'>
-                        <div
-                            className='cfd-password-modal__content dc-modal__container_cfd-password-modal__body'
-                            data-testid='dt_create_password'
-                        >
+                    <div
+                        className='cfd-password-modal__content dc-modal__container_cfd-password-modal__body'
+                        data-testid='dt_create_password'
+                    >
+                        <div className='cfd-password-modal__create-password-body mt5-migration-modal__mobile-container'>
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx]: please check the classnames. why are we using mt5-migration-modal__mobile-container classnames?

File: packages/appstore/src/components/cfds-listing/cfds-listing.scss (hunk 1 above)
Comment (Dhruv-deriv) [reply packages/appstore/src/components/cfds-listing/cfds-listing.scss]: resolved 👍 was removed by mistake

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 4 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: have cleaned , removed 👍

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 6 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: it is used in 
Screenshot 2024-08-26 at 1 18 22 PM

on testing: when removed it does not load the password modal

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx (hunk 7 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx]: removed was added as it was used in previous PR, now not needed

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx (hunk 3 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx]: done, cleaned code 👍

File: packages/cfd/src/Containers/cfd-password-modal.tsx (hunk 2 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/cfd-password-modal.tsx]: some elements are using the styles provided by mt5-create-password.scss.

Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/cfd-password-modal.tsx]: before removing import:
Screenshot 2024-08-26 at 1 37 57 PM

after removing import: 
Screenshot 2024-08-26 at 1 39 42 PM

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+        &-text {
+            border-radius: 0.8rem;
+            padding: 0.8rem;
+            background-color: var(--status-warning);
+        }
+    }
+    &__message {
+        max-width: 32rem;
+        margin: auto;
+        line-height: 1.43;
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss:51]: why no unit?

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 9]
```diff
@@ -0,0 +1,226 @@
+.dc-modal__container_cfd-password-modal,
+.dc-mobile-dialog__cfd-password-modal {
+    display: flex;
+    flex-direction: column;
+    justify-content: flex-start;
+
+    form {
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: how is this being used in component?

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+
+        & .btn {
+            min-width: 9rem;
+        }
+    }
+    &--error {
+        text-transform: none;
+    }
+    @include mobile-or-tablet-screen {
+        &--absolute {
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: are we using this?

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 9 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss:7]: removed 👍

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 8 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss:51]: have removed __message as its not being used by MT5Creaste password modal

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 9 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: without the form styling the component shrinks in mobile view 
Screenshot 2024-08-26 at 3 36 53 PM

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+        .cfd-password-modal__content {
+            display: flex;
+            flex-direction: column;
+            width: 100%;
+            margin-bottom: 2rem;
+        }
+    }
+}
+
+.cfd-password-modal {
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: lets use unique classnames

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 10 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: nope its removed

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+
+    @include mobile-or-tablet-screen {
+        .mt5-password-modal__content {
+            overflow-y: auto;
+            overflow-x: hidden;
+            height: 100%;
+            max-height: calc(100vh - 6.4rem);
+            padding: 1rem;
+        }
+    }
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: lets double check on this

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+    &__body {
+        display: flex;
+        flex-direction: column;
+        justify-content: flex-start;
+        align-items: flex-start;
+    }
+
+    .mt5-password-field {
+        margin-bottom: 1rem;
+        // width: 80%;
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: remove commented code

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 6 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: resolved 👍

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 11 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: resolved 👍

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 12 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: have removed no errros 👍

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 13 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: have removed  👍

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 8 above)
Comment (amina-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss:51]: is it done?

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 14]
```diff
@@ -0,0 +1,128 @@
+.dc-modal__container_mt5-password-modal,
+.dc-mobile-dialog__mt5-password-modal {
+    display: flex;
+    flex-direction: column;
+    padding: 4rem 1.6rem 1.6rem;
+    justify-content: flex-start;
+    @include mobile-or-tablet-screen {
+        .mt5-password-modal__content {
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: do we need this ?
why not move it to line:73 block?

Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: resolved 👍

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+            width: 100%;
+            align-items: center;
+            padding-bottom: 1rem;
+
+            @include mobile-or-tablet-screen {
+                height: 100%;
+                padding: 4rem 1.6rem;
+            }
+            .dc-icon {
+                margin-left: 0.5rem;
```
Comment (fasihali-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: Can we use logical property here  😬

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx [hunk 16]
```diff
@@ -68,21 +69,25 @@ const MT5CreatePassword = ({
...
-                            <div className='cfd-password-modal__create-password-content'>
-                                <Icon icon='IcMt5OnePassword' width='100' height='100' />
-                            </div>
-                            <Text size='xs' align='left' className='cfd-password-modal__create-password-description'>
+                    <div
+                        className='mt5-password-modal__content dc-modal__container_mt5-password-modal__body form'
+                        data-testid='dt_mt5_create_password'
+                    >
+                        <div className='mt5-password-modal__create-password-content'>
+                            <Icon icon='IcMt5OnePassword' width='150' height='150' />
```
Comment (fasihali-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx]: Should we use Quill icon here or is it fine ? 😬

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 15 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss:84]: hey @fasihali-deriv just to be clear are u referring to margin-start?

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx (hunk 16 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx]: hey @fasihali-deriv  i will clarify once

Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.tsx]: hey @fasihali-deriv we can implement quill icons here 👌

**Review by fasihali-deriv — APPROVED** [review]

File: packages/cfd/src/Containers/mt5-create-password/__tests__/mt5-create-password.spec.js [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+        render(
+            <Router history={history}>
+                <MT5CreatePassword {...default_props} />
+            </Router>,
+            {
+                wrapper: ({ children }) => <CFDProviders store={mockStore(mockRootStore)}>{children}</CFDProviders>,
+            }
+        );
+
+        expect(await screen.findByTestId('dt_create_password')).toBeInTheDocument();
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/__tests__/mt5-create-password.spec.js]: check this testid

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 18]
```diff
@@ -0,0 +1,124 @@
+.dc-modal__container_mt5-password-modal,
+.dc-mobile-dialog__mt5-password-modal {
+    display: flex;
+    flex-direction: column;
+    justify-content: flex-start;
+    @include mobile-or-tablet-screen {
+        .mt5-password-modal__content {
+            overflow-y: auto;
+            overflow-x: hidden;
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: check this classnames

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 19]
```diff
@@ -0,0 +1,124 @@
...
+
+        &-content {
+            display: flex;
+            flex-direction: column;
+            width: 100%;
+            align-items: center;
+            padding-bottom: 1rem;
+
+            @include mobile-or-tablet-screen {
+                height: 100%;
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss:74]: check these duplicates

Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss:74]: done

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 18 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: resolved

File: packages/cfd/src/Containers/mt5-create-password/__tests__/mt5-create-password.spec.js (hunk 17 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/__tests__/mt5-create-password.spec.js]: fixed

**Review by amina-deriv — APPROVED** [review]
